packet_decoder.go

```diff
@@ -27,6 +27,7 @@ type packetDecoder interface {
 	remaining() int
 	getSubset(length int) (packetDecoder, error)
 	peek(offset, length int) (packetDecoder, error) // similar to getSubset, but it doesn't advance the offset
```

Comment:
If this change work, do we still need peek method?

Reply:
There are no existing uses of `peek`. Should I remove it?

Reply:
let's wait. @bai you have any inputs?